src/commands/play.rs

```diff
@@ -145,16 +150,19 @@ async fn calculate_time_until_play(queue: &[TrackHandle], flag: &PlayFlag) -> Op
 }
 
 async fn enqueue_playlist(call: &Arc<Mutex<Call>>, uri: &str) {
-    let res = YoutubeDl::new(uri).flat_playlist(true).run().unwrap();
+    let mut child = Command::new("yt-dlp")
+        .args([uri, "--flat-playlist", "--print-json"])
+        .stdout(Stdio::piped())
+        .spawn()
+        .unwrap();
 
-    if let YoutubeDlOutput::Playlist(playlist) = res {
-        let entries = playlist.entries.unwrap();
+    if let Some(stdout) = &mut child.stdout {
+        let reader = BufReader::new(stdout);
+        let lines: Vec<String> = reader.lines().flatten().collect();
 
-        for entry in entries.iter() {
-            let url = format!(
-                "https://www.youtube.com/watch?v={}",
-                entry.url.as_ref().unwrap()
-            );
+        for line in lines.iter() {
+            let entry: Value = serde_json::from_str(line).unwrap();
```

Comment:
Can't we get away with implicit typing here (removing the `: Value`) ?

Other than that looks good but I haven't been able to test it

Reply:
Doesn't look like it, type annotations are needed in the line below.